network/p2p/inspector/validation/control_message_validation_inspector.go

```diff
@@ -117,486 +114,431 @@ func NewControlMsgValidationInspector(
+		case p2pmsg.CtrlMsgGraft:
+			err := c.inspectGraftMessages(req.Peer, req.rpc.GetControl().GetGraft(), activeClusterIDS)
 			if err != nil {
-				lg.Error().
-					Err(err).
-					Msg("could not pre-process rpc, aborting")
-				return fmt.Errorf("could not pre-process rpc, aborting: %w", err)
+				c.logAndDistributeAsyncInspectErrs(req, p2pmsg.CtrlMsgGraft, err)
+				return nil
+			}
+		case p2pmsg.CtrlMsgPrune:
+			err := c.inspectPruneMessages(req.Peer, req.rpc.GetControl().GetPrune(), activeClusterIDS)
+			if err != nil {
+				c.logAndDistributeAsyncInspectErrs(req, p2pmsg.CtrlMsgPrune, err)
+				return nil
+			}
+		case p2pmsg.CtrlMsgIWant:
+			err := c.inspectIWantMessages(req.Peer, req.rpc.GetControl().GetIwant())
+			if err != nil {
+				c.logAndDistributeAsyncInspectErrs(req, p2pmsg.CtrlMsgIWant, err)
+				return nil
 			}
 		case p2pmsg.CtrlMsgIHave:
-			// iHave specific pre-processing
-			sampleSize := util.SampleN(len(control.GetIhave()), c.config.IHaveInspectionMaxSampleSize, c.config.IHaveSyncInspectSampleSizePercentage)
-			err := c.blockingIHaveSamplePreprocessing(from, validationConfig, control, sampleSize)
+			err := c.inspectIHaveMessages(req.Peer, req.rpc.GetControl().GetIhave(), activeClusterIDS)
 			if err != nil {
-				lg.Error().
-					Err(err).
-					Msg("could not pre-process rpc, aborting")
-				return fmt.Errorf("could not pre-process rpc, aborting: %w", err)
+				c.logAndDistributeAsyncInspectErrs(req, p2pmsg.CtrlMsgIHave, err)
+				return nil
 			}
 		}
+	}
+
+	return nil
+}
 
-		// queue further async inspection
-		req, err := NewInspectMsgRequest(from, validationConfig, control)
+// inspectGraftMessages performs topic validation on all grafts in the control message using the provided validateTopic func while tracking duplicates.
+// Args:
+// - from: peer ID of the sender.
+// - grafts: the list of grafts to inspect.
+// - activeClusterIDS: the list of active cluster ids.
+// Returns:
+// - DuplicateFoundErr: if there are any duplicate topics in the list of grafts
+// - error: if any error occurs while sampling or validating topics, all returned errors are benign and should not cause the node to crash.
+func (c *ControlMsgValidationInspector) inspectGraftMessages(from peer.ID, grafts []*pubsub_pb.ControlGraft, activeClusterIDS flow.ChainIDList) error {
+	tracker := make(duplicateStrTracker)
+	for _, graft := range grafts {
+		topic := channels.Topic(graft.GetTopicID())
+		if tracker.isDuplicate(topic.String()) {
+			return NewDuplicateFoundErr(fmt.Errorf("duplicate topic found: %s", topic.String()))
+		}
+		tracker.set(topic.String())
+		err := c.validateTopic(from, topic, activeClusterIDS)
 		if err != nil {
-			lg.Error().
-				Err(err).
-				Str("peer_id", from.String()).
-				Str("ctrl_msg_type", string(ctrlMsgType)).
-				Msg("failed to get inspect message request")
-			return fmt.Errorf("failed to get inspect message request: %w", err)
+			return err
```

Comment:
```suggestion
			return fmt.Errorf("failed to validate topic %s: %w", topic, err)
```

Reply:
The error currently has all the context needed this will add some unnecessary noise to the error. Eventually, these errors are disseminated in an notification with additional context added.